cluster/log-dump/log-dump.sh

```diff
@@ -502,6 +502,7 @@ function dump_nodes_with_logexporter() {
   local -r cloud_provider="${KUBERNETES_PROVIDER}"
   local -r enable_hollow_node_logs="${ENABLE_HOLLOW_NODE_LOGS:-false}"
   local -r logexport_sleep_seconds="$(( 90 + NUM_NODES / 3 ))"
+  local -r compute_zone="${LOGEXPORTER_ZONE:-}"
 
   # Fill in the parameters in the logexporter daemonset template.
   sed -i'' -e "s@{{.LogexporterNamespace}}@${logexporter_namespace}@g" "${KUBE_ROOT}/cluster/log-dump/logexporter-daemonset.yaml"
```

Comment:
I don't think changing the template in place is a right solution.
I know that it's the way it was implemented before your changes, but it's tricky. I know that it's rare, but if someone calls log-dump.sh twice, e.g. with different extra_log_files flag value, will get unexpected result.

Intentionally changing nodeSelector multiple times makes it even worse. I think it's a good time to add copy file "logexporter-daemonset.yaml" to some other place and making changes only to that copy.

Reply:
This is a good time to remove this code from k/k repository altogether.

Reply:
I'm fine with doing a copy of that YAML, but it should be done in a separate PR

Reply:
This is happening in https://github.com/kubernetes/kubernetes/pull/94681

Reply:
Rebased the PR and made it base on top of #94681.